tls/s2n_tls13_secrets.c

```diff
@@ -625,6 +663,10 @@ S2N_RESULT s2n_tls13_secrets_get(struct s2n_connection *conn, s2n_extract_secret
     RESULT_ENSURE_LTE(secret_type, CONN_SECRETS(conn).extract_secret_type);
     RESULT_ENSURE_REF(secrets[secret_type][mode]);
 
+    /* Getting secrets requires these fields to be non-null.  */
+    RESULT_ENSURE_REF(conn->secure);
+    RESULT_ENSURE_REF(conn->secure->cipher_suite);
```

Comment:
Nit: do this at the top of the function, with the other ENSURE_REFs.

Reply:
If you do, it breaks existing unit tests which expect certain types of error codes from this function.

Reply:
There are no uses of `conn->secure` (in macro or otherwise) prior to this point in the function.

Reply:
https://github.com/aws/s2n-tls/blob/36c848d21c73557e2273f6b68e1a090ad4150235/tests/unit/s2n_tls13_secrets_test.c#L311-L316

Reply:
I considered "correcting" the unit tests -- but it seemed like the specific return codes were important.

Reply:
Probably still worth correcting. To make the existing tests pass: setting up the connection with s2n_connection_new() will solve the conn->secure being null, and then you can manually set the cipher suite to whatever you want.

Reply:
Ok done.